### PR TITLE
bump: wasm shims for vulnerabilities; align shim binary and containerd handler versions

### PR DESCRIPTION
**Pull request by alexeldeib**
this provides a stable versioning approach moving forward. we will eventually drop the historical v1 we started with. typical support will only be a few months per shim.

this also aligns us with the [upstream versioning proposal][0]

[0]: https://github.com/deislabs/containerd-wasm-shims/blob/main/proposals/2022-09-14-kubernetes-versioning.md#versioning-of-both-shim-binary-and-runtimeclass-name

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Review by devigned — APPROVED** [review]
lgtm

**Review by UtheMan — APPROVED** [review]

**Comment by alexeldeib** [comment]
need the upstream build for v0.5.0, seems like a few paths moved around

**Comment by devigned** [comment]
> need the upstream build for v0.5.0, seems like a few paths moved around

Is this something I can help with? Wasn't clear based on your comment.

File: pkg/agent/baker_test.go [hunk 1]
```diff
@@ -735,7 +735,42 @@ var _ = Describe("Assert generated customData and cseCmd", func() {
...
+    [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.spin-v0-3-0]
+      runtime_type = "io.containerd.spin-v0-3-0.v1"
+    [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.slight-v0-3-0]
+      runtime_type = "io.containerd.slight-v0-3-0.v1"
+    [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.spin-v0-5-1]
+      runtime_type = "io.containerd.spin-v0-5-1.v1"
+    [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.slight-v0-5-1]
+      runtime_type = "io.containerd.slight-v0-5-1.v1"`
+
+				Expect(containerdConfigFileContent).To(ContainSubstring(expectedShimConfig))
```
Comment (cameronmeissner) [inline pkg/agent/baker_test.go:771]: mainly just curious, why don't we assert complete equality here?

**Review by cameronmeissner — APPROVED** [review]

File: pkg/agent/baker_test.go (hunk 1 above)
Comment (alexeldeib) [reply pkg/agent/baker_test.go:771]: no real reason, the real config does have things after that, so it's not a perfect equality.

mostly I kept adding bits debugging the tabs vs spaces until it was almost complete :D

**Pull request merged**